templates/zone-master.erb

```diff
@@ -1,6 +1,6 @@
 <%-
-if @is_dynamic and @allow_update.empty?
-  raise(Puppet::ParseError, "allow_update is empty for dynamic zone '#{name}'")
+if @is_dynamic and (@allow_update.empty? and @allow_update_cidr.empty?)
+  raise(Puppet::ParseError, "Both allow_update and allow_update_cidr are empty for dynamic zone '#{name}'")
```

Comment:
Not sure if allowing both parameters to be set is wise…
I think we should allow only Array XOR string.
Also, we might is is_a?(Array) in order to re-use the standard parameter, avoiding the hassle to check one more param. Of course, doing so will change the check l51-52 in the bind::zone class.